pallets/exchange/src/lib.rs

```diff
@@ -88,51 +123,73 @@ decl_module! {
 					owner: who.clone(),
 				};
 
-				*id = id.checked_add(&One::one()).ok_or(Error::<T>::OrderIdOverflow)?;
-				
+				*id = Some(
+					order_id
+						.checked_add(&One::one())
+						.ok_or(Error::<T>::OrderIdOverflow)?,
+				);
+
 				T::Currency::reserve(base_currency_id, &who, base_amount)?;
 
 				Orders::<T>::insert(order_id, &order);
 
-				Self::deposit_event(RawEvent::OrderCreated(order_id, order));
-				Ok(())
+				Self::deposit_event(Event::OrderCreated(order_id, order));
+				Ok(().into())
 			})?;
+			Ok(().into())
 		}
 
-		#[weight = 1000]
-		fn take_order(origin, order_id: T::OrderId) {
+		#[pallet::weight(10_000)]
```

Comment:
```suggestion
		#[pallet::weight(10_000)]
		#[transactional]
```

然后就没有 `with_transaction_result` 的必要了

Reply:
fix